### PR TITLE
Fix pv patch continuously  problem

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Currently, pv would be continuously patched which is not we want. I found that there are mainly 2 problems in the current logic:

1. the selector for the tidbcluster pvc would select all the pvc including `tidbmonitor`' pvc.

```golang
case v1alpha1.TiDBClusterKind:	
     l, err = label.New().Instance(instanceName).Selector()	
     if err != nil {	
         return err	
     }
```

2.  If we didn't define the `PersistentVolumeReclaimPolicy` in the Spec, the comparing here will be not equal forever.

```golang
if pv.Spec.PersistentVolumeReclaimPolicy == policy {
    continue
}
```
### What is changed and how does it work?

Make `PersistentVolumeReclaimPolicy` as pointer and fix the selector logic.

### Does this PR introduce a user-facing change?:

```release-note
Fix the bug that the `PV` managed by Operator would be patched continuously
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go [hunk 1]
```diff
@@ -60,6 +60,10 @@ func setTidbClusterSpecDefault(tc *v1alpha1.TidbCluster) {
 		d := false
 		tc.Spec.EnablePVReclaim = &d
 	}
+	recyclePVP := corev1.PersistentVolumeReclaimRecycle
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go:63]: Suggest using retain as the default policy. WDYT @cofyc 
Retain is what we suggested in our docs and only few PVs support recycle.

File: pkg/manager/meta/reclaim_policy_manager.go [hunk 2]
```diff
@@ -90,6 +74,20 @@ func (rpm *reclaimPolicyManager) sync(kind, ns, instanceName string, isPVReclaim
 		if pvc.Spec.VolumeName == "" {
 			continue
 		}
+		l := label.Label(pvc.Labels)
+		switch kind {
+		case v1alpha1.TiDBClusterKind:
+			if !l.IsPD() && !l.IsTiDB() && !l.IsTiKV() && !l.IsTiFlash() {
```
Comment (DanielZhangQD) [inline pkg/manager/meta/reclaim_policy_manager.go:80]: Pump also uses PV.

Comment (Yisaer) [reply pkg/manager/meta/reclaim_policy_manager.go]: it's ok for me.

File: pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go (hunk 1 above)
Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/defaulting/tidbcluster.go]: updated.

**Pull request merged**